src/SAWScript/Crucible/JVM/Builtins.hs

```diff
@@ -357,6 +357,10 @@ checkRegisterCompatibility :: J.Type -> J.Type -> IO Bool
 checkRegisterCompatibility mt mt' =
   return (storageType mt == storageType mt')
 
+-- | Check two Types for register compatibility.
+registerCompatible :: J.Type -> J.Type -> Bool
+registerCompatible mt mt' = storageType mt == storageType mt'
```

Comment:
Nothing wrong with this implementation, but could also be: ```registerCompatible = (==) `on` storageType```

Wondering though if `checkRegisterCompatibility` above should be changed to `return . registerCompatible` to eliminate duplicate implementations that risk diverging in the future.

Reply:
I should probaby just get rid of the monadic version completely; I was just too lazy to rewrite the use sites of the monadic one.

Reply:
As of cf055b6b8ac45b8611f17847b165de4e8f79135c, the monadic `checkRegisterCompatibility` is gone.